pkgs/tools/security/john/default.nix

```diff
@@ -1,32 +1,17 @@
 { lib, stdenv, fetchFromGitHub, openssl, nss, nspr, libkrb5, gmp, zlib, libpcap, re2
-, gcc, python3Packages, perl, perlPackages, makeWrapper, fetchpatch
-}:
+, gcc, python3Packages, perl, perlPackages, makeWrapper, }:
 
 stdenv.mkDerivation rec {
   pname = "john";
-  version = "1.9.0-jumbo-1";
+  version = "rolling-2404";
```

Comment:
Unstable versioning should be formatted like: `1.9.0-jumbo-unstable-YYYY-MM-DD`

Reply:
are you sure? "rolling-2024" is a version name upstream uses in their unstable packages, so i went with that over a custom nixpkgs specific version name

Reply:
According #234201, we should include the preceeding version if we are packaging something untagged. `rolling-2024` does not really a "version tag", is it?

Reply:
it is a version tag upstream at https://github.com/openwall/john-packages/releases/tag/rolling-2404

i don't have a strong opinion on it either way, i can still change it when I'm able to tomorrow if you think it's really important

Reply:
Oh and looks like previous versions are also tagged https://github.com/openwall/john-packages/releases/tag/rolling-2310 https://github.com/openwall/john-packages/releases/tag/rolling-2304 

Then I am open to keeping the current version `rolling-2404`. And instead of the commit, the tag can be directly fetched I suppose? See my comment below

Reply:
that doesn't work, because the tag is in a seperate repo, upstream is a bit messy there..

they've tagged that version in their "john-packages" repo, which contains some binary builds and instructions for packaging and stuff, but it's not tagged in the actual "john" repo containing the code of the project, which is the one we fetch

that's why I've just used the same commit hash that's referenced in their package repo

Reply:
to make this a bit clearer

the upstream organization has two repos

john
john-packages

john has the actual code and some old version tags, john-packages has some scripts for packaging 

on john-packages, there is an official version tag for rolling-2404, with packaging instructions that refer back to the john repo at a specific commit hash

that is why i used the version name from john-packages and directly fetch the corresponding code from john

Reply:
Generic update scripts/bots are gonna have a hard time dealing with that. Wish there was a update script. 

If you'd like to design an update script for it, you can fetch the tag from `john-packages` repo, and somehow get the commit hash from `Created-on_2024-04-03.txt` release file. Then substitute the commit hash in the package definition. 

I'd assume that would be a lot complex to design, I am not opposed to the idea of fetching the latest commit/making it stay on bleeding edge all the time. There is `unstableGitUpdater` for that too, so bots like r-ryantm can easily update it even in a maintainer crisis.

Reply:
This package has already not been updated in like 5 years, I'd rather just get any more recent version merged than bikeshedding about update-scripts to be honest..

I fixed the other comments you had, but I would prefer to just stay on the version and method we have and not do any more big changes, something like update scripts might be a thing for a future follow-up in my opinion